src/BizHawk.Emulation.Cores/Consoles/Nintendo/Gameboy/GambatteLink.IVideoProvider.cs

```diff
@@ -19,19 +22,35 @@ public partial class GambatteLink : IVideoProvider
 
 		private readonly int[] FrameBuffer;
 
-		public int[] GetVideoBuffer() => VideoBuffer;
+		public int[] GetVideoBuffer()
+		{
+			return showAnyBorder() ? SgbVideoBuffer : VideoBuffer;
+		}
 
 		private readonly int[] VideoBuffer;
-		
+
+		private readonly int[] SgbVideoBuffer;
+
 		private int[] CreateVideoBuffer()
 		{
-			var b = new int[BufferWidth * BufferHeight];
+			var b = new int[160 * _numCores * 144];
 			for (int i = 0; i < b.Length; i++)
 			{
 				b[i] = -1; // GB/C screen is disabled on bootup, so it always starts as white, not black
 			}
 			return b;
 		}
 
+		private int[] CreateSGBVideoBuffer()
+		{
+			if (isAnySgb)
+			{
+				return new int[256 * _numCores * 224];
+			}
+			else
+			{
+				return new int[1]; // create a dummy so that the fixed statement can work regardless

```

Comment:
Something seems off here, like you might want this to be nullable or throw some sort of error if an SGBVideoBuffer was asked for when isAnySgb is false and then only call SGBVideoBuffer if you know it's true